src/visualizers/VisualizerGrid.ts

```diff
@@ -0,0 +1,965 @@
+        + 'or in some other sequence.'
+
+    //Grid variables
+    amountOfNumbers = 40000
+    currentSequenceIndex = 0
+    startingIndex = 0
+    currentNumber = 0n
+    showNumbers = false
+    preset = Preset.Custom
+    pathType = PathType.Spiral
+    resetAndAugmentByOne = false
+    squareValues = false
+    backgroundColor = BLACK
+    numberColor = WHITE
+
+    //Path variables
+    x = 0
+    y = 0
+    scalingFactor = 0
+    currentDirection = Direction.Right
+    numberToTurnAtForSpiral = 0
+    incrementForNumberToTurnAt = 1
+    whetherIncrementShouldIncrement = true
+
+    //Properties
+    propertyObjects = [
+        {
+            property: Property.Prime,
+            visualization: PropertyVisualization.Primary_Color,
+            color: RED,
+        },
+    ]
+
+    params: {[key: string]: ParamInterface} = {
+        preset: {
+            value: this.preset,
+            from: Preset,
+            displayName: 'Presets',
+            required: false,
+            description:
+                'Note: If a preset is selected, properties no longer function.',
+        },
+        amountOfNumbers: {
+            value: this.amountOfNumbers,
+            displayName: 'Amount of Numbers',
+            required: false,
+            description:
+                'Note: There is an increased lag time over 10,000 numbers',
+        },
+        startingIndex: {
+            value: this.startingIndex,
+            displayName: 'Starting Index',
+            required: false,
+            description: 'Note: This does not accept negative numbers.',
+        },
+        pathType: {
+            value: this.pathType,
+            from: PathType,
+            displayName: 'How path moves about grid',
+            required: false,
+        },
+        showNumbers: {
+            value: this.showNumbers,
+            forceType: 'boolean',
+            displayName: 'Show numbers',
+            required: false,
+            description:
+                'Note: Show_Path and Show_Numbers default to 400 numbers',
+        },
+        numberColor: {
+            value: this.numberColor,
+            forceType: 'color',
+            displayName: 'Number color:',
+            required: false,
+            visibleDependency: 'showNumbers',
+            visiblePredicate: (dependentValue: boolean) =>
+                dependentValue === true,
+        },
+        backgroundColor: {
+            value: this.backgroundColor,
+            forceType: 'color',
+            displayName: 'Background color:',
+            required: false,
+        },
+    }
+
+    constructor() {
+        super()
+        for (let i = 1; i < MAXIMUM_ALLOWED_PROPERTIES; i++) {
+            this.propertyObjects.push({
+                property: Property.None,
+                visualization: PropertyVisualization.None,
+                color: DEFAULT_COLORS[i],
+            })
+        }
+
+        for (let i = 0; i < MAXIMUM_ALLOWED_PROPERTIES; i++) {
```

Comment:
The later properties (beyond 1) are inside an indented gray box.  This visual doesn't match the heirarchy; Properties should be at the top level, and their colour choices/Visualizations individually indented.

Reply:
Hmm, the trouble here is that making a property optional automatically indents it, and property 2 is optional because it only shows when property one is set, etc. So rearranging this layout would require changes to the general setup of the parameter dialog, which is likely beyond the scope of this PR. I'd suggest we file an issue once this goes in, rather than trying to fix it here.